sklearn/model_selection/_search.py

```diff
@@ -1804,3 +1807,340 @@ def _run_search(self, evaluate_candidates):
                 self.param_distributions, self.n_iter, random_state=self.random_state
             )
         )
+
+
+class Razors:
+    """
+    Razors is a callable refit option for `GridSearchCV` and `RandomSearchCV`,
```

Comment:
Razors is not callable, so I'm a bit confused by this statement. I think this discussion belongs in the narrative documentation rather than a docstring for a single class.

Altogether I don't really see the provision of these within a class as being idiomatic within scikit-learn. We tend to have collections of things as modules (with the unusual exception of something like `GenericUnviariateSelect`).

In a similar vein, I'm not sure I understand the choice of name for `simplify`.

Reply:
> I am not convinced by the "unified" design in a single class. Why this design choice?
> 
> Could we instead have `sklearn.model_selection.selet_by_standard_error` or `sklearn.model_selection.razor_by_standard_error`? (not sure razor is a standard enough term...)
Right, so the unified design was how @amueller, @NicolasHug , and I had originally conceived the prototype. With that said, I am not attached to its single-class structure if you feel that it'd be better to modularize it. If we do go that route, it sounds like the preference would not be for its atomic contents to clutter `_search.py` module, but rather to break out the functionality into its own new module within `model_selection`, correct?

Will go ahead and update the docstring to reduce any ambiguity, while relegating the more detailed discussion to narrative documentation, which I can add to this PR (in `doc/modules` assuming the functionality will indeed be broken out into its own, refit callables, module?).

Reply:
> the preference would not be for its contents to clutter _search.py module, but rather to break out the functionality into its own new module within model_selection?

Yes, please.

>  I am not attached to its single-class structure

Either separate functions/classes, or a single class that is constructed with configuration, like `Refitter(method="rank_sum")` would be much more in keeping with the library. I've realised that separate functions will need to return callables, so that's also a bit different to everything other than `make_column_selector`.

Reply:
> > the preference would not be for its contents to clutter _search.py module, but rather to break out the functionality into its own new module within model_selection?
> 
> Yes, please.
> 

Done. 

> > I am not attached to its single-class structure
> 
> Either separate functions/classes, or a single class that is constructed with configuration, like `Refitter(method="rank_sum")` would be much more in keeping with the library. I've realised that separate functions will need to return callables, so that's also a bit different to everything other than `make_column_selector`.

I've gone ahead and separated them out into their own module (`_subselect.py`, though not 100% sure about that name either). I also decided to divorce the different strategies from the (newly named) `Refitter` class to make it less opinionated and isolate the business logic of each strategy to their own respective `by_....` callable classes. This also helped with testing, and will make it easier for other users to contribute more subselection strategies down the road.